apps/internal/base/internal/storage/storage.go

```diff
@@ -238,14 +236,10 @@ func (m *Manager) aadMetadata(ctx context.Context, authorityInfo authority.Info)
 	}
 
 	for _, metadataEntry := range discoveryResponse.Metadata {
-		metadataEntry.TenantDiscoveryEndpoint = discoveryResponse.TenantDiscoveryEndpoint
 		for _, aliasedAuthority := range metadataEntry.Aliases {
 			m.aadCache[aliasedAuthority] = metadataEntry
 		}
 	}
-	// TODO(msal): Don't understand this logic.  We query first this data, we enter all the data that
```

Comment:
was this resolved?

Reply:
Yes summarized the explanation in the PR comment. Posting it here again for reference:

There is another TODO that I have removed here : https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/apps/internal/base/internal/storage/storage.go#L246

I am attaching an [internal API review](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=%2FInstance%20Discovery%20Caching%2Fdesktop_web_caching.md&version=GBdev&_a=preview&anchor=cache-precedence) doc which explains why If our host was not detailed by the server, we just insert it using default authority values.